src/odemis/driver/autoscript_client.py

```diff
@@ -1122,9 +1122,10 @@ def __init__(
         # beam current
         # NOTE: VA is named probeCurrent to match existing API
         beam_current_info = self.parent.beam_current_info(self.channel)
+        beam_current_range = beam_current_info["range"]
         self.probeCurrent = model.FloatContinuous(
```

Comment:
Change already to a check if the length > 2, and if so, make an FloatEnumerated. Actually, to handle the corner cases when there are only 2 choices, it might be best to change the xtadapter to send in such case, instead of "range", a "choices".